src/encoding_implementations/utils.jl

```diff
@@ -0,0 +1,5 @@
+# Normalize `x::Real ∈ [minval, maxval]` to the interval [minval, maxval]. Assumes
+# `minval < maxval`. This needs to be checked at a higher level if this function is called.
+function norm_minmax(x, minval, maxval)
+    return (x - minval) / (maxval - minval)
+end
```

Comment:
why does this function exist...? Isn't it much simpler to just write `(x - minval) / (maxval - minval)` everywhere in the source code that it uses it? It is definitely much clearer. I'd say to remove this file alltogether.